cassandra-migration-spring-boot-starter/src/main/java/org/cognitor/cassandra/migration/spring/CassandraMigrationAutoConfiguration.java

```diff
@@ -28,41 +32,71 @@
 @ConditionalOnClass(Cluster.class)
 public class CassandraMigrationAutoConfiguration {
     private final CassandraMigrationConfigurationProperties properties;
+    private final Cluster cluster;
 
     @Autowired
-    public CassandraMigrationAutoConfiguration(CassandraMigrationConfigurationProperties properties) {
+    public CassandraMigrationAutoConfiguration(CassandraMigrationConfigurationProperties properties, Cluster cluster) {
         this.properties = properties;
+        this.cluster = cluster;
     }
 
-
-    @Bean
+    @Bean(initMethod = "execute")
     @ConditionalOnBean(Cluster.class)
     @ConditionalOnMissingBean(TaskChain.class)
-    public TaskChain migrationProcess(Cluster cluster) {
+    public TaskChain migrationProcess() {
+        if (properties.isEnabled()) {
+            return createMigrationTaskChain();
+        } else {
+            return createEmptyTaskChain();
+        }
+    }
+
+    private TaskChain createEmptyTaskChain() {
+        return new TaskChain();
+    }
+
+    private TaskChain createMigrationTaskChain() {
+        return new TaskChainBuilder(cluster,
+                migrationConfiguration(),
+                migrationRepository())
+                .buildTaskChain(migrationDatabase());
+    }
+
+    @Bean
+    org.cognitor.cassandra.migration.Configuration migrationConfiguration() {
         if (!properties.hasKeyspaceName()) {
             throw new IllegalStateException("Please specify ['cassandra.migration.keyspace-name'] in" +
                     " order to migrate your database");
         }
         KeyspaceDefinition keyspaceDefinition = new KeyspaceDefinition(properties.getKeyspaceName())
                 .with(properties.getReplicationStrategy());
-        return new TaskChainBuilder(cluster,
-                    new org.cognitor.cassandra.migration.Configuration(keyspaceDefinition)
-                .setChecksumValidation(properties.isChecksumValidation())
+        return new org.cognitor.cassandra.migration.Configuration(keyspaceDefinition).setChecksumValidation(properties.isChecksumValidation())
                 .setValidateOnly(properties.isChecksumValidationOnly())
                 .setRecalculateChecksum(properties.isRecalculateChecksum())
                 .setRecalculateChecksumOnly(properties.isRecalculateChecksumOnly())
                 .setConsistencyLevel(properties.getConsistencyLevel())
-                .setCreateKeyspace(properties.isCreateKeyspace()),
-                createRepository())
-                .buildTaskChain();
+                .setCreateKeyspace(properties.isCreateKeyspace());
+    }
+
+    @Bean
```

Comment:
Any reason for making this a public bean if you do not use it? The only place where you are using the Database object in the createMigrationTaskChain method, you call this method explicitly. This basically means that the bean that is supposed to be a singleton is potentially created twice. Once by invoking the method directly and once if somebody else tries to get it from the Context.